home-manager/home-manager

```diff
@@ -87,14 +86,16 @@ function doBuild() {
 }
 
 function doSwitch() {
-    local wrkdir
-    wrkdir="$(mktemp -d)"
+    local wrkdir="$(mktemp -d)"
 
-    if doBuild "$wrkdir/generation" ; then
-        "$wrkdir/generation/activate"
-    fi
+    local generation
+    local exitCode=0
+    generation=$(doBuild "$wrkdir/result") && $generation/activate || exitCode=1
```

Comment:
Hmm, actually I don't understand why it doesn't work inside the `if`. From the Bash manual:

> The shell does not exit if the command that fails is part of the command list immediately following a while or until keyword, part of the test following the if or elif reserved words, part of any command executed in a && or || list except the command following the final && or ||, any command in a pipeline but the last, or if the command's return value is being inverted with !.

In particular "part of the test following the if". Anyway, I'm happy with it the way it is, no worries :-)